tests/test_ccs.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+try:
+    from unittest.mock import patch, ANY
+except:
+    from mock import patch, ANY
+
+from tests.http_client import MinimalResponse
+from tests.test_token_cache import build_response
+
+import msal
+
+
+class TestCcsRoutingInfoTestCase(unittest.TestCase):
+
+    def test_acquire_token_by_auth_code_flow(self):
+        app = msal.ClientApplication("client_id")
+        state = "foo"
+        flow = app.initiate_auth_code_flow(
+            ["some", "scope"], login_hint="johndoe@contoso.com", state=state)
+        with patch.object(app.http_client, "post", return_value=MinimalResponse(
+                status_code=400, text='{"error": "mock"}')) as mocked_method:
+            app.acquire_token_by_auth_code_flow(flow, {"state": state, "code": "bar"})
+            self.assertEqual(
+                "UPN:johndoe@contoso.com",
```

Comment:
Characters here should be lowercase

Reply:
Thanks for spotting this topic. We may want to double check with Ryan on this (I'll ping him offline). Currently, the ["Scenarios" section of the internal design doc](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?version=GBdev&path=%2FCCS%20Routing%20Headers%2FCCSRoutingHeaders.md&_a=preview&anchor=scenarios) contains examples with `UPN:...` and `Oid:...`. I'd assume many readers would follow that sample.

Either Ryan would modify the service-side implementation to accept different cases, or we'd have to: (1) update that design doc, and (2) have different MSAL teams to examine their implementation and make changes.

Reply:
Yes. Actually, casing does not really matter but it is better to keep everything lowercase since headers are usually lowercase anyway. I confirmed this with CCS already

Reply:
To summarize our offline conversation: Travis forwarded me an email which confirmed with CCS team that the *values* of the `X-AnchorMailbox: ValueS` are also treated as case-insensitive. Glad that we concluded this topic. I'll still add a comment in this implementation to call it out.